packages/sales-api-service/src/services/__tests__/contacts.service.spec.js

```diff
@@ -44,85 +44,263 @@ describe('contacts service', () => {
+          {
+            mobilePhone: email
+          }
+        ]
+        dynamicsLib.findById.mockImplementationOnce(() => ({}))
+        dynamicsLib.findByExample.mockImplementationOnce(() => contactCRM)
+        const mockPayload = {
+          email: null
+        }
+        const permit = mockPermit()
+        const contact = await resolveContactPayload(permit, mockPayload)
+        expect(contact.mobilePhone).toEqual(email)
       }
-      const contact = await resolveContactPayload(mockPermission, mockPayload)
-      expect(contact.preferredMethodOfConfirmation).toEqual(undefined)
+    )
+
+    describe('contact does not exist in crm', () => {
+      it.each([['Letter'], ['Email'], ['Text']])(
+        'preferredMethodOfReminder is set to value of payload preferredMethodOfReminder',
+        async preferredMethodOfReminder => {
+          dynamicsLib.findById.mockImplementationOnce(() => ({}))
+          dynamicsLib.findByExample.mockImplementationOnce(() => [])
+          const mockPayload = {
+            preferredMethodOfReminder: preferredMethodOfReminder
+          }
+          const permit = mockPermit()
+          const contact = await resolveContactPayload(permit, mockPayload)
+          expect(contact.preferredMethodOfReminder.description).toEqual(preferredMethodOfReminder)
+        }
+      )
+
+      it.each([['Letter'], ['Email'], ['Text']])(
+        'preferredMethodOfConfirmation is set to value of payload preferredMethodOfConfirmation',
+        async preferredMethodOfConfirmation => {
+          dynamicsLib.findById.mockImplementationOnce(() => ({}))
+          dynamicsLib.findByExample.mockImplementationOnce(() => [])
+          const mockPayload = {
+            preferredMethodOfConfirmation: preferredMethodOfConfirmation
+          }
+          const permit = mockPermit()
+          const contact = await resolveContactPayload(permit, mockPayload)
+          expect(contact.preferredMethodOfConfirmation.description).toEqual(preferredMethodOfConfirmation)
+        }
+      )
+
+      it.each([['Letter'], ['Email'], ['Text']])(
+        'shortTermPreferredMethodOfConfirmation is set to value of payload preferredMethodOfConfirmation',
+        async preferredMethodOfConfirmation => {
+          dynamicsLib.findById.mockImplementationOnce(() => ({}))
+          dynamicsLib.findByExample.mockImplementationOnce(() => [])
+          const mockPayload = {
+            preferredMethodOfConfirmation: preferredMethodOfConfirmation
+          }
+          const permit = mockPermit()
+          const contact = await resolveContactPayload(permit, mockPayload)
+          expect(contact.shortTermPreferredMethodOfConfirmation.description).toEqual(preferredMethodOfConfirmation)
+        }
+      )
+    })
+
+    describe('contact exists in crm', () => {
+      describe('long term licence', () => {
+        it.each([['Letter'], ['Email'], ['Text']])(
+          'preferredMethodOfReminder is set to value of preferredMethodOfReminder',
```

Comment:
This will appear exactly the same for each test, i.e.:
'Letter' => 'preferredMethodOfReminder is set to value of preferredMethodOfReminder'
'Email' => 'preferredMethodOfReminder is set to value of preferredMethodOfReminder'
'Text' => 'preferredMethodOfReminder is set to value of preferredMethodOfReminder'

It'd be much better to have the actual value you're varying to be in the test name:
'Letter' => 'preferredMethodOfReminder is set to Letter'
'Email' => 'preferredMethodOfReminder is set to Email'
'Text' => 'preferredMethodOfReminder is set to Text'

Do this with your other tests too, so that the names become more descriptive of what's happening